doc/branch.rst

```diff
@@ -129,3 +127,102 @@ While we might expect this line of code to be reported as covered, the
 generator did not iterate until ``StopIteration`` is raised, the indication
 that the loop is complete. This is another case
 where adding ``# pragma: no branch`` may be desirable.
+
+Explanations and examples of missing branches
+=============================================
+
+This section shows common examples of missing branches in Python code and
+explains how coverage.py reports them.
+
+``for`` loop example
+--------------------
+
+Example::
+
+    1: items = [1]
+    2: for x in items:
+    3:     print(x)
+    4:     if x:
+    5:         print("x is true")
+    6: print("done")
+
+
+Possible branches:
+
+* ``2 -> 3`` (loop body executed)
+* ``2 -> 6`` (exit)
+* ``4 -> 5`` (if True)
+* ``4 -> 2`` (if False, backward branch)
+
+
+Case 1 — ``items`` is empty::
+
+    Missing:
+    2 -> 3
+    4 -> 2
```

Comment:
You have to actually run these examples to see what coverage produces:
```
% cat empty.py
items = []
for x in items:
    print(x)
    if x:
        print("x is true")
print("done")

% coverage run --branch empty.py
done

% coverage report -m
Name       Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------
empty.py       6      3      4      1    40%   3-5
------------------------------------------------------
TOTAL          6      3      4      1    40%
```
Here the report only says `3-5` because none of those lines are run at all. Branches that are missing because the target line isn't run aren't reported as missing because that would create too much noise.